app/services/visit_timeline.rb

```diff
@@ -24,7 +24,7 @@ def initialize(visit)
   end
 
   def events
-    events = @visit.visit_state_changes.reverse.map.with_index { |state, i|
+    events = visit_state_changes.reverse.map.with_index { |state, i|
```

Comment:
I think is a bit more explicit that we reverse the ordering to be able to spot the last element